pkg/util/workqueue/timed_queue.go

```diff
@@ -39,24 +39,28 @@ func NewTimedWorkQueue(clock clock.Clock) *TimedWorkQueue {
 // Add adds the obj along with the current timestamp to the queue.
 func (q TimedWorkQueue) Add(obj interface{}) {
 	start := q.clock.Now()
```

Comment:
This kind of object is dangerous if you don't include something like:

```go
if t, ok := obj.(*TimedWorkQueueItem); ok {
  q.Type.Add(t)
  return
}
```

Then you can delete the AddWithTimestamp method, if you want.